packages/react-components/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -7,17 +7,27 @@ import { ListboxContext } from '../../contexts/ListboxContext';
 import type { OptionValue } from '../../utils/OptionCollection.types';
 import type { OptionProps, OptionState } from './Option.types';
 
-function getValueString(value: string | undefined, children: React.ReactNode) {
-  if (value) {
-    return value;
+function getTextString(text: string | undefined, children: React.ReactNode) {
+  if (text) {
```

Comment:
What if someone sets `text` to an empty string? Should we honor that value?
```suggestion
  if (text !== undefined) {
```